service/raft/raft_group0.cc

```diff
@@ -984,11 +985,15 @@ with_timeout(abort_source& as, db::timeout_clock::duration d, F&& fun) {
 static future<> wait_until_every_peer_joined_group0(db::system_keyspace& sys_ks, const group0_members& members0, abort_source& as) {
     static constexpr auto retry_period = std::chrono::seconds{1};
 
-    while (true) {
+    for (;; co_await sleep_abortable(retry_period, as)) {
         // We fetch both config and peers on each iteration; we don't assume that they don't change.
         // No new node should join while the procedure is running, but nodes may leave.
 
         auto current_config = members0.get_inet_addrs();
+        if (current_config.empty()) {
+            // Not all addresses are known
```

Comment:
Please replace the multiple `debug` calls inside `get_inet_address` (one log message for every processed member) with a single `info` call that prints the set of members for which we're still missing IPs. Look at the other `info` calls in this function. They strive to give as much useful info as possible to help investigation if the function hangs in production (where debug level logging will probably be disabled)

Reply:
We don't get to know all the members for which we're missing addresses in get_inet_addrs().

Reply:
`members0.get_inet_addrs` could return them, but nvm.